### PR TITLE
chore(docs): Improves UX for docs editors

### PR DESCRIPTION
**Pull request by mrclay**
Thanks to @Srokap this adds a bunch of cool stuff:
- grunt script to launch freshly-built docs in a browser with auto-refresh as you make changes
- RST docs to instruct setting this up
- renamed version of the ReadTheDocs sphinx theme (so we can start customizing it)

### CODE REVIEW

**Comment by Srokap** [comment]
I cleaned up the configuration here: https://github.com/Srokap/Elgg/commit/b058fe6628fc319f4cc6fad7679170e1fec32323

You basically need to run:
```
npm install
grunt
```

And it will build and watch docs. I changed it to not clean everything on every load as it's very slow. It only cleans stuff initially and than builds only changed parts. It reloads in ~4s for me.

File: docs/conf.py [hunk 1]
```diff
@@ -92,9 +92,20 @@
 
 # -- Options for HTML output ---------------------------------------------------
 
+# https://github.com/snide/sphinx_rtd_theme#using-this-theme-locally-then-building-on-read-the-docs
+#
+# on_rtd is whether we are on readthedocs.org, this line of code grabbed from docs.readthedocs.org
+on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
```
Comment (Srokap) [inline docs/conf.py]: Why not always use local theme? We did wanted to tune it up a bit to have more of look and feel of community (obviously not identical layout)

**Comment by mrclay** [comment]
@Srokap I integrated your latest commit. Thanks for the help!

**Comment by ewinslow** [comment]
This is causing a problem:

```
WARNING: favicon file 'favicon.ico' does not exist
```

**Comment by Srokap** [comment]
Fix to incorporate as well: https://github.com/Srokap/Elgg/commit/7813fe92ee64dbf98a0c7d77870e37ed9f773cf0

Works for me for both RTD and local cases.

**Comment by mrclay** [comment]
@Srokap integrated and squashed

**Comment by mrclay** [comment]
I calrified the [license](https://github.com/mrclay/Elgg-leaf/blob/rtd_theme/docs/_themes/elgg_rtd_theme/LICENSE)

**Pull request merged**